be/src/exec/pipeline/scan/scan_operator.cpp

```diff
@@ -17,12 +17,13 @@ namespace starrocks::pipeline {
 // ========== ScanOperator ==========
 
 ScanOperator::ScanOperator(OperatorFactory* factory, int32_t id, int32_t driver_sequence, ScanNode* scan_node)
-        : SourceOperator(factory, id, scan_node->name(), scan_node->id(), driver_sequence),
-          _scan_node(scan_node),
-          _chunk_source_profiles(MAX_IO_TASKS_PER_OP),
-          _is_io_task_running(MAX_IO_TASKS_PER_OP),
-          _chunk_sources(MAX_IO_TASKS_PER_OP) {
-    for (auto i = 0; i < MAX_IO_TASKS_PER_OP; i++) {
+        : SourceOperator(factory, id, scan_node->name(), scan_node->id(), driver_sequence), _scan_node(scan_node) {
+    int io_tasks_per_scan_operator = scan_node->io_tasks_per_scan_operator();
+    _chunk_source_profiles.resize(io_tasks_per_scan_operator);
+    _is_io_task_running.resize(io_tasks_per_scan_operator);
```

Comment:
`_is_io_task_running` is `std::vector<std::atomic_bool>`, which is not allowed `resize`, but only initialized with size. Therefore, maybe move it to the initialized list of `ScanOperator::ScanOperator()`